docs: Spelling corrections

mcauser opened the pull request:
paramter -> parameter
send a receive -> send and receive
respones -> response
succeses -> success
repsonse -> response
particualr -> particular
constructore -> constructor

dpgeorge commented on cc3200/hal/prcm.c:
```diff
@@ -1015,7 +1015,7 @@ void PRCMSRAMRetentionDisable(unsigned long ulSramColSel, unsigned long ulFlags)
 //! \param ulHIBWakupSrc is logical OR of valid HIB wakeup sources.
 //!
-//! This function enables individual HIB wakeup source(s). The paramter
+//! This function enables individual HIB wakeup source(s). The parameter
```
Please don't change this file.  It's from an upstream source.  You can try getting TI to change it if you really want :)

dpgeorge commented on drivers/nrf24l01/nrf24l01test.py:
```diff
@@ -57,7 +57,7 @@ def master():
         pyb.delay(250)
 
-    print('master finished sending; succeses=%d, failures=%d' % (num_successes, num_failures))
+    print('master finished sending; success=%d, failures=%d' % (num_successes, num_failures))
```
This should be "successes"

dpgeorge reviewed (commented):
Thanks, but please split it into 2 commits: one for the docs files and another for the drivers files.

mcauser committed:
docs: Spelling corrections

paramter -> parameter
send a receive -> send and receive
repsonse -> response
particualr -> particular
constructore -> constructor

mcauser committed:
drivers: Spelling corrections

respones -> response
succeses -> successes

mcauser commented:
Bump

dpgeorge commented:
Sorry, I didn't notice that you responded to my comments.  Now merged, see c5310ee5b545316d7b8770343908c19f1ff1cd0d.
